test/start-stop.node.js

```diff
@@ -188,6 +188,58 @@ tests.forEach((fOpts) => {
       })
     })
 
+    describe('should detect and attach to running node', () => {
+      let ipfsd
+      let exec
+
+      before(function (done) {
+        this.timeout(50 * 1000)
+
+        const df = IPFSFactory.create(dfConfig)
+        exec = findIpfsExecutable(fOpts.type)
+
+        df.spawn({
+          exec,
+          initOptions: { bits: fOpts.bits }
+        }, (err, daemon) => {
+          expect(err).to.not.exist()
+          expect(daemon).to.exist()
+
+          ipfsd = daemon
+          done()
+        })
+      })
+
+      after((done) => ipfsd.stop(done))
+
+      it('should return a node', () => {
+        expect(ipfsd).to.exist()
+      })
+
+      it('shoul attach to running node', function (done) {
+        this.timeout(50 * 1000)
+
+        const df = IPFSFactory.create(dfConfig)
+        df.spawn({
+          initOptions: { bits: fOpts.bits },
+          repoPath: ipfsd.repoPath,
+          disposable: false
+        }, (err, daemon) => {
+          expect(err).to.not.exist()
+          daemon.start((err, api) => {
+            expect(err).to.not.exist()
+            expect(api).to.exist()
+            expect(ipfsd.apiAddr).to.be.eql(daemon.apiAddr)
+            daemon.stop(done)
```

Comment:
necessary to `stop`? The `after` hook does it and if we've made it this far through the expects, `ipfsd` == `daemon`.

Reply:
`daemon` refers to the second started node.

Reply:
ah! So two daemons communicating to the same remote ipfs. Thanks